api/src/main/java/org/apache/iceberg/catalog/Catalog.java

```diff
@@ -46,7 +46,8 @@ default String name() {
    *
    * @param namespace a namespace
    * @return a list of identifiers for tables
-   * @throws NotFoundException if the namespace is not found
```

Comment:
`NotFoundException` is used for file-based operations and no catalog is throwing it. 
Hence, updating it based on the current implementations.